src/conn/conn_sweep.c

```diff
@@ -266,7 +268,8 @@ __sweep_server(void *arg)
 		WT_ERR(__sweep_mark(session, &dead_handles));
 
 		if (dead_handles == 0 &&
-		    conn->open_file_count < conn->sweep_handles_min)
+		    conn->open_file_count < conn->sweep_handles_min &&
```

Comment:
I think this should be:

```
if (dead_handles == 0 &&
    (conn->open_file_count < conn->sweep_handles_min ||
    conn->sweep_idle_time != 0))
```

i.e: We never want to skip closing handles if we see any dead handles.
We want to skip closing handles if either there aren't enough open, or we've disabled closing idle handles.

The test case should be expanded to capture this:
Setting ```sweep_handles_min``` relatively low, opening and closing lots of handles, and ensuring they never get closed if ```sweep_idle_time``` is set to 0.